aspnetcore/fundamentals/file-providers/samples/1.x/FileProviderSample/Controllers/HomeController.cs

```diff
@@ -14,13 +15,10 @@ public HomeController(IFileProvider fileProvider)
 
         public IActionResult Index()
         {
-            var contents = _fileProvider.GetDirectoryContents("");
-            return View(contents);
-        }
+            var contents = _fileProvider.GetDirectoryContents(string.Empty);
```

Comment:
This kinda seems like a really "meh" sample. I wouldn't recommend doing this simply because StaticFileMiddleware \ DirectoryBrowserMiddleware would be the recommended approach to doing this. Also unrelated, there's like 90k+ lines of unrelated css, js files. Are those actually needed here?

Reply:
@pranavkm 

On the lines, *yes.* We're including *lib* folder assets to make sure our samples run OOB on a download of the repo. There might be a plan to stop doing this after we get LibMan; but for now, this is what we're doing.

On this approach, I merely stuck with what's there. I agree with what you're suggesting, but we're really under-the-:gun: right now to get these samples updated to 2.1. How about I open an issue to create a better concept demo sample for this topic? ... something that isn't a dup of what Static File Middleware and Directory Browser Middleware would do. I can probably work it within a couple of sprints.

Reply:
Opened ... *Create a better File Providers topic sample* #7874

Reply:
> something that isn't a dup of what Static File Middleware and Directory Browser Middleware would do.

IMO, just adding it to the file providers in static file provider might be good enough? I don't think `IFileProvider` is a very good general purpose abstraction for users to use as much as it is a way for Asp.Net Core components to work with file like things.

Reply:
Actually ... let me *devil's advocate* on this for a sec with what we currently have: Static File Middleware and Directory Browser Middleware don't get one directory contents or `IFileInfo` **_to work with_**. This sample doesn't look at only `wwwroot` ... it goes for files in the app's content root. I admit that the directory listing leans simplistic ... but ... but ... we're really just trying to show the dev how to get the `IDirectoryContents`. The other sample example that watches for changes seems good ... clear ... easy to build something from (I mean as a starting concept of how to call `Watch` and use the `IChangeToken`).

Yes ... simple examples ... but this doesn't seem like complicated subject matter. Once the dev sees they're getting an `IDirectoryContents`, `IFileInfo`, or `IChangeToken`, they should be fine from there for whatever they plan to use this for.

Now, you can accuse the *devil* of picking *very gooberly advocates!* :smile: lol